pose-detection/src/calculators/tracker.ts

```diff
@@ -72,17 +76,62 @@ export abstract class Tracker {
   }
 
   /**
-   * Performs an optimization to link detections with tracks. The `poses`
+   * Performs a greedy optimization to link detections with tracks. The `poses`
    * array is updated in place by providing an `id` property. If incoming 
    * detections are not linked with existing tracks, new tracks will be created.
-   * @param poses An array of detected `Pose's.
+   * @param poses An array of detected `Pose's. It's assumed that poses are
+   * sorted from most confident to least confident.
    * @param simMatrix A 2D array of shape [num_det, num_tracks] with pairwise
    * similarity scores between detections and tracks.
    * @param timestamp The current timestamp in milliseconds.
    */
   assignTracks(
       poses: Pose[], simMatrix: number[][], timestamp: number): void {
-    //TODO: Implement optimization and track store mechanics.
+    const unmatchedTrackIndices = Array.from(Array(simMatrix[0].length).keys());
+    const detectionIndices = Array.from(Array(poses.length).keys());
+    const matchedDetectionIndices: number[] = [];
+
+    for (const detectionIndex of detectionIndices) {
+      if (!unmatchedTrackIndices) {
+        break;
+      }
+
+      // Assign the detection to the track which produces the highest pairwise
+      // similarity score, assuming the score exceeds the minimum similarity
+      // threshold.
+      let maxTrackIndex = -1;
+      let maxSimilarity = -1;
+      for (const trackIndex of unmatchedTrackIndices) {
+        const similarity = simMatrix[detectionIndex][trackIndex];
+        if (similarity >= this.minSimilarity && similarity > maxSimilarity) {
+          maxTrackIndex = trackIndex;
+          maxSimilarity = similarity;
+        }
+      }
+      if (maxTrackIndex >= 0) {
+        // Link the detection with the highest scoring track.
+        this.tracks[maxTrackIndex].lastTimestamp = timestamp;
+        this.tracks[maxTrackIndex].keypoints = (
+          poses[detectionIndex].keypoints.slice());
+        const index = unmatchedTrackIndices.indexOf(maxTrackIndex);
+        unmatchedTrackIndices.splice(index, 1);
+        matchedDetectionIndices.push(detectionIndex);
+      }
+    }
+
+    // Spawn new tracks for all unmatched detections.
+    for (const detectionIndex of detectionIndices) {
+      if (matchedDetectionIndices.includes(detectionIndex)) {
```

Comment:
You could also keep track of unmatched detection indices instead of matched detection indices, so you can directly loop over the unmatched detection indices to spawn new tracks, instead of looping over all detection indices and skipping all matched ones.